File: OpenEdXMobile/src/main/java/org/edx/mobile/view/login/LoginPresenter.java
package org.edx.mobile.view.login;

import androidx.annotation.NonNull;

import org.edx.mobile.util.Config;
import org.edx.mobile.view.ViewHoldingPresenter;

public class LoginPresenter extends ViewHoldingPresenter<LoginPresenter.LoginViewInterface> {

    final private Config config;

    public LoginPresenter(Config config) {
        this.config = config;
    }

    @Override
    public void attachView(@NonNull LoginViewInterface view) {
        super.attachView(view);
        view.setSocialLoginButtons(config.getGoogleConfig().isEnabled(),
                config.getFacebookConfig().isEnabled(),
                config.getMicrosoftConfig().isEnabled());

        if (!config.isRegistrationEnabled()) {
            view.disableToolbarNavigation();
        }
    }

    public interface LoginViewInterface {

        void setSocialLoginButtons(boolean googleEnabled, boolean facebookEnabled, boolean microsoftEnabled);

        void disableToolbarNavigation();
    }
}

File: OpenEdXMobile/src/main/java/org/edx/mobile/view/login/LoginPresenter.kt
package org.edx.mobile.view.login

import org.edx.mobile.util.Config
import org.edx.mobile.view.ViewHoldingPresenter
import org.edx.mobile.view.login.LoginPresenter.LoginViewInterface

class LoginPresenter constructor(private val config: Config) :
    ViewHoldingPresenter<LoginViewInterface>() {

    override fun attachView(view: LoginViewInterface) {
        super.attachView(view)
        view.setSocialLoginButtons(
            googleEnabled = config.googleConfig.isEnabled,
            facebookEnabled = config.facebookConfig.isEnabled,
            microsoftEnabled = config.microsoftConfig.isEnabled
        )
        if (!config.isRegistrationEnabled) {
            view.disableToolbarNavigation()
        }
    }

    interface LoginViewInterface {
        fun setSocialLoginButtons(
            googleEnabled: Boolean,
            facebookEnabled: Boolean,
            microsoftEnabled: Boolean
        )

        fun disableToolbarNavigation()
    }
}
